datalad/customremotes/base.py

```diff
@@ -635,16 +646,26 @@ def get_URLS(self, key):
                 url = url[1:]
                 if url:
                     assert(len(url) == 1)
-                    urls.append(url[0])
+                    nurls += 1
+                    scheme_urls.append(url[0])
                 else:
                     break
+            if scheme_urls:
+                # note: generator would ceise to exist thus not asking
+                # for URLs for other schemes if this scheme is good enough
```

Comment:
s/ceise/cease/

I don't understand this comment and how it relates to the commit where it was added (7da0b6909).

Reply:
it somewhat relates to both changes of 1. making it a generator; 2. sorting so we hit first the one we had succeeded before with; and probably to 1. more than to the 2.  But since both commits are in the same PR I guess no need to move the comment, unless you insist.
As for more explanation re 1.: since it is a generator, if the first scheme succeeds to find URLs to which we e.g. perform successfully the CHECKPRESENT, then we would not even consider the next scheme/prefix since generator will not be asked for the "next" one.  If it was not a generator -- we go through all schemes regardless.

Reply:
> since it is a generator, if the first scheme succeeds to find URLs to which we e.g. perform successfully the CHECKPRESENT, then we would not even consider the next scheme/prefix since generator will not be asked for the "next" one. If it was not a generator -- we go through all schemes regardless.

If I understand correctly, perhaps something like this then: ""Note: Yield a list of URLs per scheme so that a caller can decide to stop when it gets a scheme that it considers good enough".